cardano-node/src/Cardano/Node/Run.hs

```diff
@@ -105,6 +106,23 @@ runNode
   :: PartialNodeConfiguration
   -> IO ()
 runNode cmdPc = do
+    -- Workaround to ensure that the main thread throws an async exception on
+    -- receiving a SIGTERM signal.
+#ifdef UNIX
+    -- Similar implementation to the RTS's handling of SIGINT (see GHC's
+    -- TopHandler.hs)
+    runThreadIdWk <- mkWeakThreadId =<< myThreadId
+    void $ Signals.installHandler
+      Signals.sigTERM
+      (Signals.CatchOnce $ do
+        runThreadIdMay <- deRefWeak runThreadIdWk
+        case runThreadIdMay of
+          Nothing -> return ()
+          Just runThreadId -> killThread runThreadId
```

Comment:
Should we also put some output here signalling that the thread was successfully killed?

Reply:
This code was lifted from GHC. We use a weak pointer so that the main thread can be garbage collected. If `doRefWeak` returns `Nothing`, then the main thread must have already exited. I can only imagine that might be possible if the main thread had shut down by some other means and a SIGTERM happened just after that. In that case I'd expect the process to exit gracefully any way, so simply doing nothing here (`return ()`) sounds correct. Is that convincing or do you still think we should output some message?

Reply:
^^ Fair points. I think it's fine as is.